internal/datastore/common/revisions.go

```diff
@@ -38,12 +41,17 @@ func (rcr *RemoteClockRevisions) OptimizedRevision(ctx context.Context) (datasto
 	defer span.End()
 
 	localNow := time.Now()
+	rcr.RLock()
 	if localNow.Before(rcr.revisionValidThrough) {
 		log.Debug().Time("now", localNow).Time("valid", rcr.revisionValidThrough).Msg("returning cached revision")
+		rcr.RUnlock()
 		return rcr.lastQuantizedRevision, nil
 	}
+	rcr.RUnlock()
 
 	lastQuantizedRevision, err, _ := rcr.updateGroup.Do("", func() (interface{}, error) {
+		rcr.Lock()
+		defer rcr.Unlock()
 		log.Debug().Time("now", localNow).Time("valid", rcr.revisionValidThrough).Msg("computing new revision")
 
 		nowHLC, err := rcr.NowFunc(ctx)
```

Comment:
Does this mean the DB access is under the lock? If so, we can't have that occur.

Reply:
refactored so that this is not an issue anymore